frontend/src/pages/projects/screens/detail/notebooks/useNotebookImageData.ts

```diff
@@ -24,24 +24,83 @@ export const getNotebookImageData = (
   }
 
   const [imageName, versionName] = imageTag;
-  const [lastImageSelectionName] =
+  const [lastImageSelectionName, lastImageSelectionTag] =
     notebook.metadata.annotations?.['notebooks.opendatahub.io/last-image-selection']?.split(':') ??
     [];
 
-  // Fallback for non internal registry clusters
-  const imageStream =
-    images.find((image) => image.metadata.name === imageName) ||
-    images.find((image) =>
-      image.spec.tags
-        ? image.spec.tags.find(
-            (version) =>
-              version.from?.name === container.image &&
-              image.metadata.name === lastImageSelectionName,
-          )
-        : false,
-    );
-
-  // if the image stream is not found, consider it deleted
+  const notebookImageInternalRegistry = getNotebookImageInternalRegistry(
+    notebook,
+    images,
+    imageName,
+    versionName,
+  );
+  if (
+    notebookImageInternalRegistry &&
+    notebookImageInternalRegistry.imageAvailability !== NotebookImageAvailability.DELETED
+  ) {
+    return notebookImageInternalRegistry;
+  }
+  const notebookImageNoInternalRegistry = getNotebookImageNoInternalRegistry(
+    notebook,
+    images,
+    lastImageSelectionName,
+    versionName,
+    imageName,
```

Comment:
```suggestion
    container.image,
```

Delete versionName cause it's not needed and it wasn't imageName, it was container.image